Monorepo

plmercereau opened the pull request.

elitan commented on examples/react-apollo-crm/package.json:
```diff
@@ -1,15 +1,15 @@
-    "@nhost/nhost-js": "^0.3.4",
-    "@nhost/react-apollo": "^2.0.7-0",
-    "@nhost/react-auth": "^2.0.3",
+    "@nhost/nhost-js": "workspace:*",
+    "@nhost/react-apollo": "workspace:*",
+    "@nhost/react-auth": "workspace:*",
```
How will this work for developers who copy the `react-apollo-crm` folder to a separate folder and runs this example independently?

plmercereau replied:
corrected. this syntax is only useful to generate version range when publishing packages to npm. Examples are not published as packages therefore don't need this.

elitan commented on package.json:
```diff
@@ -0,0 +1,78 @@
+{
+  "name": "nhost-js-root",
+  "private": true,
+  "version": "0.0.0",
+  "description": "Nhost Javascript SDK",
+  "author": "Nhost",
+  "license": "MIT",
+  "homepage": "https://nhost.io",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nhost/nhost-js.git"
+  },
+  "bugs": {
+    "url": "https://github.com/nhost/nhost-js/issues"
+  },
```
Do we need to update the name, description, and URLs for this package.json?

plmercereau replied:
I'm not fond on this `-js` suffix to be honest. Any npm package is meant to be Javascript, this suffix is redundant. I did not want to change the package name as we are not talking about renaming our existing npm packages, although I think we should

elitan replied:
Yep I see your point and I agree.

The question is if we want to introduce the mismatch now? Or keep things in sync for now until we make such name change?

plmercereau replied:
I haven't changed the package name and can keep add `-js` suffixes to the folders although it is hurting my eyes :)

elitan replied:
I don't have a particular hard opinion here tbh. Maybe @nunopato wants to chime in?

elitan replied:
> I haven't changed the package name and can keep add `-js` suffixes to the folders although it is hurting my eyes :)

At least we're being consistent! :D

plmercereau replied:
ah ok, I saw your second comment about the root package.json which is somehow incorrect. Pushing changes

elitan reviewed (changes requested):
The folder name of e.g. `@nhost/hasura-auth-js` is `hasura-auth` (with out the `-js`). 

Should we instead try to keep the folder and package names in sync?

Like this:

`@nhost/nhost-js` -> `packages/nhost-js`
`@nhost/hasura-auth-js` -> `packages/hasura-auth-js`

etc

elitan commented:
Will packages' versions be updated individually with this monorepo setup?

elitan commented:
There are a lot of config files in the root directory of the repo.

Is there any chance we can reduce those? Fewer files in the root directory mean less scrolling to the README file :)

![CleanShot 2022-02-07 at 09 43 48@2x](https://user-images.githubusercontent.com/331818/152754197-4e89dbdf-ef67-4fcf-a00d-29ebd1f66fe7.png)

plmercereau commented:
> 


> Will packages' versions be updated individually with this monorepo setup?

Yes, that's the spirit

plmercereau commented:
> There are a lot of config files in the root directory of the repo.
> 
> Is there any chance we can reduce those? Fewer files in the root directory mean less scrolling to the README file :)
> 
> ![CleanShot 2022-02-07 at 09 43 48@2x](https://user-images.githubusercontent.com/331818/152754197-4e89dbdf-ef67-4fcf-a00d-29ebd1f66fe7.png)

yes, that's too much. I moved this to a `.config` folder that is populated when the `prepare` script is executed, and added the root files to `.gitignore`
